cpp_linter/clang_tools/patcher.py

```diff
@@ -0,0 +1,67 @@
+"""A module to contain the abstractions about creating suggestions from a diff generated
+by the clang tool's output."""
+
+from abc import ABC
+from typing import Optional, Dict, Any
+from unidiff import PatchSet
+from ..common_fs import FileObj
+from ..loggers import logger
+
+
+class PatchMixin(ABC):
+    """An abstract mixin that unified parsing of the suggestions into
+    PR review comments."""
+
+    def __init__(self) -> None:
+        #: A unified diff of the applied fixes from the clang tool's output
+        self.patched: Optional[str] = None
+
+    def get_suggestion_help(self, start, end) -> str:
+        """Create helpful text about what the suggestion aims to fix."""
+
+        raise NotImplementedError("derivative must implement this abstract method")
+
+    def get_suggestions_from_patch(
+        self,
+        file_obj: FileObj,
+        summary_only: bool,
+    ):
+        """Create a list of suggestions from the tool's `patched` output."""
+
+        if not self.patched:
+            logger.debug(
+                "%s has no suggestions for %s", self.__class__.__name__, file_obj.name
+            )
+            return ("", [], 0)
```

Comment:
This is no longer an assertion because the generated diff of suggested fixes is saved to `TidyAdvice.patched` or `FaormatAdvice.patched`. If there is no suggested fixes, then [`difflib.unified_diff()`](https://docs.python.org/3/library/difflib.html#difflib.unified_diff) returns an empty string. See also changes in [clang-tidy.py](https://github.com/cpp-linter/cpp-linter/pull/115/files#diff-f633f1dfb198669ddc2086ce6ac1e9817803a4ed775e5065d95f74bcd08d6f80) and [clang-format.py](https://github.com/cpp-linter/cpp-linter/pull/115/files#diff-5310aec821f0c94a8f355ea0a3e8fb389aa78764a93f8f483b8a3649defcc88f).